modules/groovy/src/main/java/org/jpos/transaction/participant/GroovyParticipant.java

```diff
@@ -116,7 +116,7 @@
     @SuppressWarnings("unchecked")
     public int prepare (long id, Serializable ctx) {
         if (prepare == null)  {
-            return PREPARED | NO_JOIN | READONLY; // nothing to do
+            return PREPARED | READONLY; // nothing to do
```

Comment:
a little optimization would be return `NO_JOIN`  if `abort`  and `commit` are null, although  it doesn't have much sense unless only `prepareForAbort` is defined

Reply:
Good point. Not a big difference, performance-wise, but good to keep in mind.